process/block/metablock.go

```diff
@@ -864,9 +864,12 @@ func (mp *metaProcessor) requestMissingFinalityAttestingHeaders() uint32 {
 				mp.dataPool.HeadersNonces())
 
 			if err != nil {
-				requestedBlockHeaders++
-				go mp.onRequestHeaderHandlerByNonce(shardId, i)
-				continue
+				shardHeader, shardHeaderHash, err = mp.getShardHeaderWithNonce(shardId, i)
```

Comment:
Is this an edge case when a header is evicted from the nonce pool, but it exists in the header pool ? It looks like an ugly hack, maybe the real error is somewhere else.

Reply:
Yes, this is the case because we use the same HeaderNonce pool either for meta headers and for shard headers and because of bootstrap this pool is filled with Shard Headers and Meta Headers are evicted after you bootstrap 1000 shar headers. Instead the MetaBlocks pool will still have them as it is dedicated only for meta blcoks, and in this time while you bootstrap 1000 shard headers you probably have around 20 new meta blocks in.

Reply:
But the real problem I found it today and is because we don't evict shard headers committed from pool because of an old approach which was used to quickly response to requests. Now anyway the resolver is looking first in the storage(cahcer) so we don't nedd to keep them there. So I will replace this implementation today with some few lines of remove old code :).